experimental/gcp-log/internal/storage/storage.go

```diff
@@ -403,10 +435,23 @@ func (c *Client) StoreTile(ctx context.Context, level, index uint64, tile *api.T
 
 	// Pass an empty rootDir since we don't need this concept in GCS.
 	tPath := filepath.Join(layout.TilePath("", level, index, tileSize%256))
+
+	// Check that either we are writing the tile for the first time, or that the
```

Comment:
How gnarly would it be to only perform this assertion when we've concretely hit the precondition fail on `Close()`? Just thinking it'd save a lot of RPCs if we don't have to check every single file up front.

Reply:
Good point! It actually also made the logic handling the return value of `assertContent` a little simpler because we eliminate the case of the object not existing.